pkg/connector/source.go

```diff
@@ -282,13 +288,39 @@ func (s *Source) state() SourceState {
 
 func (s *Source) configure(ctx context.Context) error {
 	s.Instance.logger.Trace(ctx).Msg("configuring source connector plugin")
-	err := s.plugin.Configure(ctx, s.Instance.Config.Settings)
+	_, err := s.plugin.Configure(ctx, pconnector.SourceConfigureRequest{Config: s.Instance.Config.Settings})
 	if err != nil {
 		return cerrors.Errorf("could not configure source connector plugin: %w", err)
 	}
 	return nil
 }
 
+func (s *Source) open(ctx context.Context) error {
+	s.Instance.logger.Trace(ctx).Msg("opening source connector plugin")
+	_, err := s.plugin.Open(ctx, pconnector.SourceOpenRequest{
+		Position: s.state().Position,
+	})
+	if err != nil {
+		return cerrors.Errorf("could not open source connector plugin: %w", err)
+	}
+	return nil
+}
+
+func (s *Source) run(ctx context.Context) error {
+	s.Instance.logger.Trace(ctx).Msg("running source connector plugin")
+	ctx, s.stopStream = context.WithCancel(ctx)
+	stream := s.plugin.NewStream()
+	err := s.plugin.Run(ctx, stream)
+	if err != nil {
+		s.stopStream()
+		s.stopStream = nil
+		s.stream = nil
```

Comment:
Hmm, in which cases would `s.stream` be non-nil? Only if `run()` was already called?

Reply:
Good point, this is redundant.